packages/postgres/README.md

```diff
@@ -156,6 +156,41 @@ const { user, course } = await sqldb.runInTransactionAsync(async () => {
 
 `runInTransaction` will start a transaction and then execute the provided function. Any nested query will use the same client and thus run inside the transaction. If the function throws an error, the transaction is rolled back; otherwise, it is committed.
 
+### Cursors
+
+For very large queries that don't need to fit in memory all at once, it's possible to use a cursor to read a limited number of rows at a time.
+
+```ts
+import { queryCursor } from '@prairielearn/postgres';
+
+const cursor = await queryCursor(sql.select_all_users, {});
+for await (const users of cursor.iterate(100)) {
+  // `users` will have up to 100 rows in it.
+  for (const user of users) {
+    console.log(user);
+  }
+}
+```
+
+You can optionally pass a Zod schema to parse and validate each row:
+
+```ts
+import { z } from 'zod';
+import { queryCursor } from '@prairielearn/postgres';
+
+const UserSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+});
+
+const cursor = await queryCursor(sql.select_all_users, {}, UserSchema);
```

Comment:
Rather than a separate `queryValidatedCursor` function, I opted to just make the model an optional third argument. This made sense to me because the function will yield pure arrays instead of a results object, so we can treat the validated case as basically the same.

Reply:
Ahh, I buy that rationale (though in theory we can *also* write a lint rule to detect 2-argument calls to `queryCursor`. I'll make this change regardless.

Reply:
Changed in 8e5d612.